src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.BlazorWasmDebuggingExtension/src/extension.ts

```diff
@@ -25,46 +25,51 @@ export function activate(context: vscode.ExtensionContext) {
-            for await (const output of spawnedProxy.stdout) {
-                // If we haven't found the URL in the first ten chunks processed
-                // then bail out.
-                if (chunksProcessed++ > 10) {
-                    return;
+            try {
+                let chunksProcessed = 0;
+                for await (const output of spawnedProxy.stdout) {
+                    // If we haven't found the URL in the first ten chunks processed
+                    // then bail out.
+                    if (chunksProcessed++ > 10) {
+                        return;
+                    }
+                    outputChannel.appendLine(output);
+                    // The debug proxy server outputs the port it is listening on in the
+                    // standard output of the launched application. We need to pass this URL
+                    // back to the debugger so we extract the URL from stdout using a regex.
+                    // The debug proxy will not exit until killed via the `killDebugProxy`
+                    // method so parsing stdout is necessary to extract the URL.
+                    const matchExpr = 'Now listening on: (?<url>.*)';
+                    const found = `${output}`.match(matchExpr);
+                    const url = found?.groups?.url;
+                    if (url) {
+                        outputChannel.appendLine(`Debugging proxy is running at: ${url}`);
+                        pidsByUrl.set(url, spawnedProxy.pid);
+                        return {
+                            url,
+                            inspectUri: `${url}{browserInspectUriPath}`,
+                            debuggingPort,
+                        };
+                    }
                 }
-                outputChannel.appendLine(output);
-                // The debug proxy server outputs the port it is listening on in the
-                // standard output of the launched application. We need to pass this URL
-                // back to the debugger so we extract the URL from stdout using a regex.
-                // The debug proxy will not exit until killed via the `killDebugProxy`
-                // method so parsing stdout is necessary to extract the URL.
-                const matchExpr = 'Now listening on: (?<url>.*)';
-                const found = `${output}`.match(matchExpr);
-                const url = found?.groups?.url;
-                if (url) {
-                    outputChannel.appendLine(`Debugging proxy is running at: ${url}`);
-                    pidsByUrl.set(url, spawnedProxy.pid);
-                    return {
-                        url,
-                        inspectUri: `${url}{browserInspectUriPath}`,
-                        debuggingPort,
-                    };
+
+                for await (const error of spawnedProxy.stderr) {
+                    outputChannel.appendLine(`ERROR: ${error}`);
+                    process.kill(spawnedProxy.pid);
```

Comment:
Just to confirm, all of these extra process.kills are the fix correct?

Reply:
These extra process.kills are targeted toward the spawned debug proxy. I don't believe I've seen any issues specifically calling out the debug proxy being kept alive past debugging termination (but due to the nature of how the port is accessed, it's possible the proxy is kept alive in the background silently). Regardless, these error states seemed like good places to also terminate the proxy as it'll no longer be functional.

The fix for the app server not being killed (and thus the port remaining open) is over in the [`BlazorDebugConfigurationProvider`](https://github.com/dotnet/razor-tooling/pull/5885/files#diff-786b1c9872b99de0589bae6ced551a7822dc6d335f86987ad55b4ac600f52d37R31). It's composed of better matching the event name (we were trying string equality with `Blazor Debug...` with `Blazor Debug...: https://localhost:7291` which failed). As well as improved REGEX matching of the `process.cmd` that we were trying to kill. The REGEX pattern contained the unescaped directory path (with unescaped `/`), which meant we weren't able to find the process to kill within our process list.

Reply:
Ahhh the StartsWith, thank you for confirming!